### PR TITLE
(DOCSP-16089): Draft section for running a password reset func, add Bluehawked example

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-16089

### Staged Changes (Requires MongoDB Corp SSO)

- [Manage Email Password Users -> Run a Password Reset Function (New section)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-16089/sdk/ios/examples/manage-email-password-users/#run-a-password-reset-function)
- [Manage Email Password Users -> Reset a User's Password (new verbiage above code examples)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-16089/sdk/ios/examples/manage-email-password-users/#reset-a-user-s-password)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/ios/Examples/ManageEmailPasswordUsers.swift [hunk 1]
```diff
@@ -104,4 +104,35 @@ class ManageEmailPasswordUsers: XCTestCase {
         // :code-block-end:
         wait(for: [expectation], timeout: 10)
     }
+
+    func testPasswordResetFunc() {
+        let expectation = XCTestExpectation()
+        // :code-block-start: password-reset-function
+        let app = App(id: YOUR_REALM_APP_ID)
```
Comment (nathan-contino-mongo) [inline examples/ios/Examples/ManageEmailPasswordUsers.swift:111]: thought/suggestion: I omit the App initialization from many Android examples to try to keep code blocks to a minimum digestible size. Not required at all but if you like the idea, feel free to do so.

File: examples/ios/Examples/ManageEmailPasswordUsers.swift [hunk 2]
```diff
@@ -104,4 +104,35 @@ class ManageEmailPasswordUsers: XCTestCase {
...
+        let client = app.emailPasswordAuth
+
+        let email = "forgot.my.password@example.com"
+        let newPassword = "mynewpassword12345"
+        // The password reset function takes any number of
+        // arguments. You might ask the user to provide answers to
+        // security questions, for example, to verify the user
+        // should be able to complete the password reset.
```
Comment (nathan-contino-mongo) [inline examples/ios/Examples/ManageEmailPasswordUsers.swift:119]: Or maybe an even easier example -- maybe another function could send a code to a user's email address, and this function could accept that code? (or perhaps that's too close to the passwordresetemail flow)

File: source/sdk/ios/examples/manage-email-password-users.txt [hunk 3]
```diff
@@ -63,6 +63,27 @@ Confirm a New User's Email Address
...
+
+- Send a password reset email
+- Run a password reset function
+
+Select your preferred password reset method by going to:
+
+- Your {+app+}
+- :guilabel:`Authentication`
+- :guilabel:`Authentication Providers`
+- :guilabel:`Email/Password` - and press the :guilabel:`EDIT` button
```
Comment (nathan-contino-mongo) [inline source/sdk/ios/examples/manage-email-password-users.txt:76]: Should these be numbered steps since they are ordered?

File: source/sdk/ios/examples/manage-email-password-users.txt [hunk 4]
```diff
@@ -63,6 +63,27 @@ Confirm a New User's Email Address
...
+
+Send a Password Reset Email
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+After you configure your {+app+} to :ref:`send a password reset email 
+<auth-send-a-password-reset-email>`, you can call ``sendResetPasswordEmail`` 
+with the user's email. {+backend+} sends an email to the user containing
```
Comment (nathan-contino-mongo) [inline source/sdk/ios/examples/manage-email-password-users.txt:83]:
```suggestion
with the user's email. {+backend+} sends an email to the user that contains 
```
Reads easier to me. Not a big deal at all though.

File: source/sdk/ios/examples/manage-email-password-users.txt [hunk 5]
```diff
@@ -76,3 +97,29 @@ Reset a User's Password
 
       .. literalinclude:: /examples/generated/code/start/ManageEmailPasswordUsers.codeblock.reset-password-objc.m
          :language: objectivec
+
+Run a Password Reset Function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you configure your app to :ref:`run a password reset function 
+<auth-run-a-password-reset-function>`, you'll define the function that 
+should run when you call ``callResetPasswordFunction()`` from the SDK.
```
Comment (nathan-contino-mongo) [inline source/sdk/ios/examples/manage-email-password-users.txt:106]: Should we link to the API doc here?

File: source/sdk/ios/examples/manage-email-password-users.txt [hunk 6]
```diff
@@ -76,3 +97,29 @@ Reset a User's Password
...
          :language: objectivec
+
+Run a Password Reset Function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you configure your app to :ref:`run a password reset function 
+<auth-run-a-password-reset-function>`, you'll define the function that 
+should run when you call ``callResetPasswordFunction()`` from the SDK. 
+This function can take a username, a password, and any number of 
+additional arguments. You can use these arguments to specify things like
```
Comment (nathan-contino-mongo) [inline source/sdk/ios/examples/manage-email-password-users.txt:108]: For some reason, "things like" sounds super informal to me. Maybe "details like"?

File: source/sdk/ios/examples/manage-email-password-users.txt [hunk 7]
```diff
@@ -76,3 +97,29 @@ Reset a User's Password
...
+When you configure your app to :ref:`run a password reset function 
+<auth-run-a-password-reset-function>`, you'll define the function that 
+should run when you call ``callResetPasswordFunction()`` from the SDK. 
+This function can take a username, a password, and any number of 
+additional arguments. You can use these arguments to specify things like 
+security question answers or other challenges that the user should pass 
+to successfully complete a password reset.
+
+You might prefer to use a custom password reset function when you want to
+define your own password reset flows. This might include things like
```
Comment (nathan-contino-mongo) [inline source/sdk/ios/examples/manage-email-password-users.txt:113]: more things, maybe we can reword this to avoid the "things like" phrasing again

Comment (nathan-contino-mongo) [reply source/sdk/ios/examples/manage-email-password-users.txt:113]: perhaps "implementations such as"?

**Review by nathan-contino-mongo — APPROVED** [review]
LGTM with some small nits/suggestions/thoughts

File: examples/ios/Examples/ManageEmailPasswordUsers.swift (hunk 1 above)
Comment (dacharyc) [reply examples/ios/Examples/ManageEmailPasswordUsers.swift:111]: Hmm. All of the other iOS examples on this page show the app initialization. I guess I assumed that was a convention we had adopted intentionally for sync apps to help orient the reader that they're looking at a sync example vs. a local example.

I think I'll leave it here so it's consistent with the other examples on this page, but it's something I'll ponder bigger-picture as I look at our catalog of iOS examples.

File: examples/ios/Examples/ManageEmailPasswordUsers.swift (hunk 2 above)
Comment (dacharyc) [reply examples/ios/Examples/ManageEmailPasswordUsers.swift:119]: Security questions and a cooldown period are the examples used on the Sync-side version of this page. I added this to be consistent with what a user would find if they clicked through to that page. 

The flow you're suggesting - having another function send a code to a user's email address, and having this function accept the code - makes total sense. But it does feel like that's added complexity - i.e. you have to have or at least explain two functions, versus a single function that could accept as a param some sort of security question answer to verify user identity.

File: source/sdk/ios/examples/manage-email-password-users.txt (hunk 3 above)
Comment (dacharyc) [reply source/sdk/ios/examples/manage-email-password-users.txt:76]: Yup, good catch, thank you!

File: source/sdk/ios/examples/manage-email-password-users.txt (hunk 5 above)
Comment (dacharyc) [reply source/sdk/ios/examples/manage-email-password-users.txt:106]: Good call, thanks for the suggestion!

File: source/sdk/ios/examples/manage-email-password-users.txt (hunk 6 above)
Comment (dacharyc) [reply source/sdk/ios/examples/manage-email-password-users.txt:108]: I don't mind informal, but "details" is probably a better fit for this specific instance.

File: source/sdk/ios/examples/manage-email-password-users.txt (hunk 7 above)
Comment (dacharyc) [reply source/sdk/ios/examples/manage-email-password-users.txt:113]: "Implementations such as" is a bit of a mouthful. I'll reword it to remove some gerunds for Nick, improve precision and simplify:

> For example, you might send a custom password reset email from a specific domain, or through a service other 
than email.
